ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -208,6 +213,16 @@ def _router_bgp_vrfs(self) -> dict | None:
                 if address_families:
                     bgp_vrf["address_families"] = address_families
 
+                for node_item in (mc_node_settings := default(get(vrf, "l3_multicast.node_settings"), get(tenant, "l3_multicast.node_settings"), [])):
+                    if self._hostname in (mc_nodes := get(node_item, "nodes", required=False, default=[])) or not mc_nodes:
+                        if not mc_nodes and len(mc_node_settings) > 1:
+                            raise AristaAvdMissingVariableError(
+                                f"l3_multicast.node_settings in Tenant '{tenant['name']}' or VRF '{vrf['name']}': only one entry with no 'nodes' or multiple"
+                                " entries with 'nodes' can be defined."
+                            )
+                        bgp_vrf["evpn_multicast_address_family"] = {"ipv4": get(node_item, "evpn_peg", default={})}
+                        break
```

Comment:
Thanks, much clearer.

Re the error class - none of the other error classes in errors.py fit - it's not a schema validation error, nor is it the problem with the AVD code - it's a data error.  I could add another class but it would be a duplicate of the MissingVariableError (just a pass statement).

Reply:
Just use `AristaAvdError` if nothing else matches.

Reply:
Fixed this.